feat: terminal light and dark mode selection

cdrage opened the pull request:
feat: terminal light and dark mode selection

### What does this PR do?

* Adds the ability to select your terminal theme from dark or light

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->


https://github.com/user-attachments/assets/3a90d27f-1b86-4663-b430-1cea26e79ed9



### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop/issues/7545

We will follow up with:
https://github.com/containers/podman-desktop/issues/8190 and
https://github.com/containers/podman-desktop/issues/8174 in the future

### How to test this PR?

<!-- Please explain steps to verify the functionality,
do not forget to provide unit/component tests -->

- [X] Tests are covering the bug fix or the new feature

1. Go to Terminal preferences and set your theme
2. Go to your Pod logs / Terminal, or anywhere else in PD that has a
   terminal
3. See the new theme.

**NOTE:** If you are going back to the "same" section and it's cached
(ex. terminal), you will have to refresh the screen.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

cdrage commented:
Please note that this is the initial implementation and we will have to follow up with refactoring + another minor feature in the future: https://github.com/containers/podman-desktop/issues/8190 and
https://github.com/containers/podman-desktop/issues/8174

deboer-tim reviewed (commented):
I really like the change and choice of a few themes, it's really slick.

The fact that the change is only picked up by some windows and not others is really confusing to the user though, especially since it's one of the only preferences like that - it's easy to miss the description and wonder why it's not working correctly. Have you checked whether it's possible to force theme updates, or maybe we should just throw out our terminal caches when this setting changes?

cdrage commented:
> IMHO it looks confusing to have terminal theming option.
> 
> 
> 
> I would expect to select a global appearance theme, not having something specific for terminals. 
> 
> 
> 
> For example it means I could have a light terminal with a dark theme or the opposite.
> 
> 
> 
> To me, I would expect that the colors of the terminal would be provided by the definition of our global colors. And if you want to customize, you would install an extension providing the theme. Here it is limiting the extensibility by hardcoding available themes in Podman Desktop. While it should be contributed by 3rd party extensions

How about just dark and light option and no extra themes? So we could at least implement it for light mode / have the option.

cdrage commented on packages/main/src/plugin/terminal-theme.ts:
```diff
@@ -0,0 +1,207 @@
+  background: '#ffffff',
+};
+
+export const themes: { [key: string]: ITheme } = {
```
changed! to match the other sections, i've changed it to lowecase now.

cdrage commented on packages/renderer/src/lib/compose/ComposeDetailsLogs.svelte:
```diff
@@ -119,14 +119,15 @@ async function refreshTerminal() {
   const lineHeight = await window.getConfigurationValue<number>(
     TerminalSettings.SectionName + '.' + TerminalSettings.LineHeight,
   );
+  const terminalTheme = await window.getConfigurationValue<string>(
```
I initially did this, but found it did not "sync" when updating. So I added this to match the other parts of the terminal.

I opened up #8174 for when we refactor Terminal

cdrage commented:
@benoitf @deboer-tim

I've updated the PR with a new commit that:

- Only adds dark and light. I agree with @benoitf after thinking about it. If we added some "common" themes like Zenburn, Monokai, it may introduce debates / discussions on what colours are the "right ones", etc. Since there is so many different combinations / how everyone views the colours differently. So may as well just stick to the default dark / light.
- Switches to lowercase light / dark in description

@deboer-tim regarding the "automatic" clear cache / updating terminal. I could not find an elegant solution without beginning to refactor every terminal page, so instead let's update it in #8174 ?

benoitf commented:
> How about just dark and light option and no extra themes? So we could at least implement it for light mode / have the option.

>The reason for selective preference is so you can have a dark terminal despite selecting light theme. As users like me may > want to have PD in light mode but terminals dark for visibility.

I don't think there should be a separated theming option just for Terminal.
A theme include all options including the terminal rendering.

so, if you select a light theme you'll get a light mode terminal and if you select a dark theme you'll get a dark mode terminal.

if you want to have different colors, then you'll provide a custom theme that would override colors of the terminal extending/overriding a given theme.

deboer-tim commented:
We had a brief chat on the standup today and can continue tomorrow on UX, but dropping my thoughts here in the meantime. I agree with Florent that the best way to deliver this for now (which might be easier as a clean PR/branch) is just  adding variables to the color registry, one set of light/dark options, and using those with no separate preference. It works, follows our extensibility pattern, gives a consistent theme with one setting, and will be an easy review/merge.

Personally, yes, I think it would be nice to allow more options for the terminal without requiring every theme to be extended with every terminal option, but that discussion is separate and going to take longer to conclude one way or the other, so we should track it separately.

I'm still hesitant to deliver without the ability to update terminal colors live and not knowing when #8174 might drop (might be nice to confirm we have a technical solution or start by fixing just one view like Logs), but yes, that is a separate issue too.

cdrage commented:
> We had a brief chat on the standup today and can continue tomorrow on UX, but dropping my thoughts here in the meantime. I agree with Florent that the best way to deliver this for now (which might be easier as a clean PR/branch) is just adding variables to the color registry, one set of light/dark options, and using those with no separate preference. It works, follows our extensibility pattern, gives a consistent theme with one setting, and will be an easy review/merge.
> 
> Personally, yes, I think it would be nice to allow more options for the terminal without requiring every theme to be extended with every terminal option, but that discussion is separate and going to take longer to conclude one way or the other, so we should track it separately.
> 
> I'm still hesitant to deliver without the ability to update terminal colors live and not knowing when #8174 might drop (might be nice to confirm we have a technical solution or start by fixing just one view like Logs), but yes, that is a separate issue too.

I agree and thank you for talking about it! I'll close this for now and tackle it in a different direction (more integrated, use color registry, use current pattern), as well as see if we can do a live update to the colours.
